### PR TITLE
[hotfix]: fix a bug in CI command

### PR DESCRIPTION
**Pull request by min-xu-ai**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by min-xu-ai** [comment]
A incoming PR uncovered a bug in the CI config: when import torch fails, the command should return OK so that we go ahead with pip installation. This version should fixes it.

cc @vfdev-5

**Comment by msbaines** [comment]
> A incoming PR uncovered a bug in the CI config: when import torch fails, the command should return OK so that we go ahead with pip installation. This version should fixes it.
> 
> cc @vfdev-5

Can you elaborate. Why should we continue if import torch fails? Isn't that catastrophic.

**Comment by min-xu-ai** [comment]
> Can you elaborate. Why should we continue if import torch fails? Isn't that catastrophic.

Still testing. Converted to draft.

The check here is that we want to continue to do the pip install of torch if it doesn't exist.

The old code is that we do always do pip install even if restoring from the cache was successful. That can fail from time to time due to networking issues.

I first put in a check for the right torch version to skip the pip install but that check was buggy. It skipped install when torch is not there.

**Comment by min-xu-ai** [comment]
This is ready. The failure is unrelated. I tested on CI and verified that we skip pip install when cache is correctly restored and with right torch version.

File: .circleci/config.yml [hunk 1]
```diff
@@ -65,22 +65,23 @@ install_dep_151: &install_dep_151
...
+        wget -O /home/circleci/venv/check_version.py https://raw.githubusercontent.com/min-xu-ai/check_verion/main/check_version.py
 
 install_dep_160: &install_dep_160
   - run:
       name: Install Dependencies with torch 1.6.0
       command: |
         sudo apt-get install -y libopenmpi-dev
         # check if we have restored venv cache (/home/circleci/venv) correctly, if so, just skip
-        python -c 'import torch; assert torch.__version__.split(".")[:2] != ["1", "6"]' || exit 0
+        if [ -f /home/circleci/venv/check_version.py ]; then python /home/circleci/venv/check_version.py torch eq 1.6 && exit 0; fi
```
Comment (blefaudeux) [inline .circleci/config.yml:84]: alrighty, makes sense

File: .circleci/config.yml [hunk 2]
```diff
@@ -146,8 +150,8 @@ install_dep_180: &install_dep_180
         pip install --progress-bar off  git+https://github.com/min-xu-ai/torch_pg.git@c723ab4#egg=torch-pg
         python -c 'import torch; print("Torch version:", torch.__version__)'
         python -c 'import torch; assert torch.__version__.split(".")[:2] == ["1", "8"], "wrong torch version"'
-        pip list|grep torch
         python -m torch.utils.collect_env
+        wget -O /home/circleci/venv/check_version.py https://raw.githubusercontent.com/min-xu-ai/check_verion/main/check_version.py
```
Comment (blefaudeux) [inline .circleci/config.yml:154]: typo, no ? 'check_verion'

File: .circleci/config.yml [hunk 3]
```diff
@@ -432,14 +436,14 @@ jobs:
       # Cache the venv directory that contains dependencies
       - restore_cache:
           keys:
-            - cache-key-gpu-151-{{ checksum "setup.py"}}-{{ checksum "requirements-test.txt"}}
+            - cache-key-gpu-151-1-{{ checksum "setup.py"}}-{{ checksum "requirements-test.txt"}}
```
Comment (blefaudeux) [inline .circleci/config.yml:439]: given the setup.py change, this is not needed, is it ?

**Comment by blefaudeux** [comment]
LGTM except for a couple of questions, mind checking Min ? Thanks in any case, I realize that this is a bit arduous but needed

File: .circleci/config.yml (hunk 2 above)
Comment (min-xu-ai) [reply .circleci/config.yml:154]: actually, it is not. It is how I made my repo. (by accident)

File: .circleci/config.yml (hunk 3 above)
Comment (min-xu-ai) [reply .circleci/config.yml:439]: right. I can remove it

**Comment by min-xu-ai** [comment]
> LGTM except for a couple of questions, mind checking Min ? Thanks in any case, I realize that this is a bit arduous but needed

Checked. Good call on the "-1-" part. Will remove.

**Review by blefaudeux — COMMENTED** [review]
LGTM with the additional read, thanks again Min !

**Pull request closed without merging**